src/ApiClient/RecommendationsClient.php

```diff
@@ -20,7 +22,14 @@ public function list(
         bool $descendingOrder = true
     ) : PromiseInterface {
         return $this->getRequest(
-            'recommendations/'.$type.'/'.$id.'?page='.$page.'&per-page='.$perPage.'&order='.($descendingOrder ? 'desc' : 'asc'),
+            Uri::fromParts([
+                'path' => "recommendations/$type/$id",
+                'query' => build_query(array_filter([
```

Comment:
this seems like standard boilerplate that every Client has to endure. Maybe we need a small abstraction but seems overdesigned